### PR TITLE
Add hourly_to_daily process

### LINKED ISSUE #185: Resample to daily process
Stumbled upon this problem while writing up the earthcube notebook. 

For a _truly_ remote computation with "arbitrary" data, it would be nice that finch provides some kind of "resampling" process. I am currently thinking a something that would take subdaily inputs and aggregate it to daily using a specific function (min, max, mean, sum). That would allow chain-computations with more datasets!

### PR DESCRIPTION
**Pull request by huard**
## Overview

This PR fixes #185

Changes:

* Added hourly_to_daily process reducing hourly data to the daily frequency using mean, sum, max or min.

### CODE REVIEW

File: finch/processes/subset.py [hunk 1]
```diff
@@ -24,6 +24,18 @@
...
 
 
+def make_subset_file_name(resource, kind="sub"):
+    """Create output file name."""
+    if resource.prop == "file":
+        p = Path(resource.file)
+    elif resource.prop == "url":
+        p = Path(resource._build_file_name(resource.url))
+    else:
+        raise NotImplementedError
```
Comment (Zeitsperre) [inline finch/processes/subset.py:34]:
```suggestion
        raise NotImplementedError()
```

File: finch/processes/subset.py [hunk 2]
```diff
@@ -24,6 +24,18 @@
...
+
+    return p.stem + f"_{kind}" + p.suffix
```
Comment (Zeitsperre) [inline finch/processes/subset.py:36]:
```suggestion
    return f"{p.stem}_{kind}{p.suffix}"
```

File: environment.yml [hunk 3]
```diff
@@ -22,8 +22,8 @@ dependencies:
...
   # remember to match xclim version in environment-docs.yml as well
   - xclim == 0.31
-  - clisops >=0.6.4
+  - clisops >= 0.8.0
   - pywps ==4.4.5
   - parse
-  - xesmf >=0.5.3,!=0.6.1  # clisops 0.7 bugs with xESMF 0.6.1
-  - ESMpy <= 8.1.1 # Bug in ESMF 8.2 with spatial averaging.
\ No newline at end of file
+  - xesmf == 0.6.2  # clisops 0.7 bugs with xESMF 0.6.1
```
Comment (aulemahal) [inline environment.yml:28]:
```suggestion
  - xesmf >= 0.6.2
```
So we are allowed an update of xesmf before updating finch? And to replicate the requirements file.

File: finch/processes/wps_hourly_to_daily.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        kls.validate(**missing_options)
+
+    # Resample to daily
+    out = getattr(ds.resample(time="D"), reducer)(keep_attrs=True)
+
+    # Update and format attributes
+    for key, da in out.data_vars.items():
+        # Update cell_methods
+        da.attrs["cell_methods"] = (
+            da.attrs.get("cell_methods", " ") + f" time: {reducer}"
```
Comment (aulemahal) [inline finch/processes/wps_hourly_to_daily.py:105]: Should we add something like "within days"?

Comment (huard) [inline finch/processes/wps_hourly_to_daily.py:105]:
```suggestion
            da.attrs.get("cell_methods", " ") + f" time: {reducer} within days"
```

Comment (huard) [reply finch/processes/wps_hourly_to_daily.py]: done

**Pull request merged**